### DISCUSSION TITLE
.NET Core Docker containers & allowedVersions

### DISCUSSION THREAD

**Discussion by jokay** [renovatebot/renovate#26641] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.131.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### What is your question?

Given a repo having a Dockerfile containing `mcr.microsoft.com/dotnet/sdk:8.0.100-alpine3.18@sha256:e6a2272325d3ec6c8f82cd0ebaa6a25104d24147e38edd9fd80b0918e4b2b2df` and the following config:

```json
   "stabilityDays": 3,
   {
      "groupName": ".NET Core Docker containers",
      "matchDatasources": ["docker"],
      "matchPackagePrefixes": ["mcr.microsoft.com/dotnet/"],
      "allowedVersions": "<9"
    }
```

I would expect it to propose the update to **8.0.101-alpine3.18** or even **8.0.101-alpine3.19**.

### Logs (if relevant)

<details><summary>Logs</summary>
DEBUG: Falling back to npm semver syntax for allowedVersions (repository=test/repo1)
       "depName": "mcr.microsoft.com/dotnet/sdk"

DEBUG: Falling back to npm semver syntax for allowedVersions (repository=test/repo1)
       "depName": "mcr.microsoft.com/dotnet/sdk"

DEBUG: Falling back to npm semver syntax for allowedVersions (repository=test/repo1)
       "depName": "mcr.microsoft.com/dotnet/aspnet"

DEBUG: packageFiles with updates (repository=test/repo1, baseBranch=main)
       "config": {
         "dockerfile": [
           {
             "deps": [
               {
                 "depName": "mcr.microsoft.com/dotnet/sdk",
                 "currentValue": "8.0.100-alpine3.18",
                 "currentDigest": "sha256:e6a2272325d3ec6c8f82cd0ebaa6a25104d24147e38edd9fd80b0918e4b2b2df",
                 "replaceString": "mcr.microsoft.com/dotnet/sdk:8.0.100-alpine3.18@sha256:e6a2272325d3ec6c8f82cd0ebaa6a25104d24147e38edd9fd80b0918e4b2b2df",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "stage",
                 "updates": [],
                 "packageName": "mcr.microsoft.com/dotnet/sdk",
                 "versioning": "docker",
                 "warnings": [],
                 "sourceUrl": "https://github.com/dotnet/sdk",
                 "registryUrl": "https://mcr.microsoft.com",
                 "currentVersion": "8.0.100",
                 "fixedVersion": "8.0.100-alpine3.18"
               },
               {
                 "depName": "mcr.microsoft.com/dotnet/aspnet",
                 "currentValue": "8.0.0-alpine3.18",
                 "currentDigest": "sha256:24d9700c32b29b7215bf7ec669dd034eb74e3935e7051e45c24d95de446bd066",
                 "replaceString": "mcr.microsoft.com/dotnet/aspnet:8.0.0-alpine3.18@sha256:24d9700c32b29b7215bf7ec669dd034eb74e3935e7051e45c24d95de446bd066",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "final",
                 "updates": [],
                 "packageName": "mcr.microsoft.com/dotnet/aspnet",
                 "versioning": "docker",
                 "warnings": [],
                 "sourceUrl": "https://github.com/dotnet/aspnetcore",
                 "registryUrl": "https://mcr.microsoft.com",
                 "currentVersion": "8.0.0",
                 "fixedVersion": "8.0.0-alpine3.18"
               }
             ],
             "packageFile": "src/app/Dockerfile"
           }
         ],
         "gitlabci": [
           {
             "packageFile": ".gitlab-ci.yml",
             "deps": [
               {
                 "depName": "mcr.microsoft.com/dotnet/sdk",
                 "currentValue": "8.0.100-alpine3.18",
                 "currentDigest": "sha256:e6a2272325d3ec6c8f82cd0ebaa6a25104d24147e38edd9fd80b0918e4b2b2df",
                 "replaceString": "mcr.microsoft.com/dotnet/sdk:8.0.100-alpine3.18@sha256:e6a2272325d3ec6c8f82cd0ebaa6a25104d24147e38edd9fd80b0918e4b2b2df",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "image",
                 "updates": [],
                 "packageName": "mcr.microsoft.com/dotnet/sdk",
                 "versioning": "docker",
                 "warnings": [],
                 "sourceUrl": "https://github.com/dotnet/sdk",
                 "registryUrl": "https://mcr.microsoft.com",
                 "currentVersion": "8.0.100",
                 "fixedVersion": "8.0.100-alpine3.18"
               },
               {
                 "depName": "docker.io/oxsecurity/megalinter-dotnet",
                 "currentValue": "v7.7.0",
                 "currentDigest": "sha256:64efa1f4445dcf0618ee25992ebec13c4be6b0c988c626f3cec05a1ef5dd5fde",
                 "replaceString": "docker.io/oxsecurity/megalinter-dotnet:v7.7.0@sha256:64efa1f4445dcf0618ee25992ebec13c4be6b0c988c626f3cec05a1ef5dd5fde",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "image",
                 "updates": [],
                 "packageName": "docker.io/oxsecurity/megalinter-dotnet",
                 "versioning": "docker",
                 "warnings": [],
                 "sourceUrl": "https://github.com/oxsecurity/megalinter",
                 "registryUrl": "https://index.docker.io",
                 "homepage": "https://megalinter.io",
                 "currentVersion": "v7.7.0",
                 "fixedVersion": "v7.7.0"
               }
           }
         ]
       }
</details>

**Comment by viceice** [comment, marked as answer]
you need custom regex versioning to support that. current docker[^1] versioning is treating `-alpine3.18` as compabillity, so if needs to be the same on all versions.

[^1]: https://docs.renovatebot.com/modules/versioning/#docker-versioning
